tests/python/gaia-ui-tests/gaiatest/apps/homescreen/app.py

```diff
@@ -172,9 +152,23 @@ def bookmark(self, bookmark_title):
 
     @property
     def number_of_columns(self):
-        element = self.marionette.find_element(*self._landing_page_locator)
-        Wait(self.marionette).until(lambda m: element.get_attribute('cols') is not None)
-        return int(element.get_attribute('cols'))
+
+        _icon_container_text = 'div.gaia-container-child:nth-child(%s)'
+        change_row = False
+        #first element is always on the first column
+        _first_icon_locator = (By.CSS_SELECTOR, _icon_container_text % str(1))
+        base_x_axis = self.marionette.find_element(*_first_icon_locator).rect['x']
+
+        # loop until the row is changed (i.e., x value is same as the first icon)
+        column = 1
+        while not change_row:
```

Comment:
Nit: add `is` or `has` for a boolean variable. For example: `has_row_changed` or `is_on_new_row`